pkg/devconsole/tekton-results/results.go

```diff
@@ -0,0 +1,299 @@
+package tektonresults
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/devconsole/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var (
+	TektonResultsResource = &schema.GroupVersionResource{
+		Group:    "operator.tekton.dev",
+		Version:  "v1alpha1",
+		Resource: "tektonresults",
+	}
+	TektonResultsAPIRoute = &schema.GroupVersionResource{
+		Group:    "route.openshift.io",
+		Version:  "v1",
+		Resource: "routes",
+	}
+	cachedTektonResultsHost string = ""
+)
+
+func isLocalhost(r *http.Request) (bool, error) {
+	host := r.RemoteAddr
+	switch {
+	case host == "::1":
+		return true, nil
+	case host == "[::1]":
+		return true, nil
+	}
+	h, p, err := net.SplitHostPort(host)
+
+	// addrError helps distinguish between errors of form
+	// "no colon in address" and "too many colons in address".
+	// The former is fine as the host string need not have a
+	// port. Latter needs to be handled.
+	addrError := &net.AddrError{
+		Err:  "missing port in address",
+		Addr: host,
+	}
+	if err != nil {
+		if err.Error() != addrError.Error() {
+			return false, err
+		}
+		// host string without any port specified
+		h = host
+	} else if len(p) == 0 {
+		return false, errors.New("invalid host name format")
+	}
+
+	// use ipv4 dotted decimal for further checking
+	if h == "localhost" {
+		h = "127.0.0.1"
+	}
+	ip := net.ParseIP(h)
+
+	return ip.IsLoopback(), nil
+}
+
+func getTRHost(r *http.Request, dynamicClient *dynamic.DynamicClient) (string, error) {
+	// If running on localhost, use the route to get the host.
+	// Required for local development.
+	localHost, err := isLocalhost(r)
```

Comment:
I think this would be better as a flag that passed to bridge on startup for local dev like how we handle the metrics endpoints. For example:

https://github.com/openshift/console/blob/master/cmd/bridge/main.go#L460-L480

It would be better to avoid logic as in `isLocalhost` if we can.